### ISSUE TITLE
Support Go's "`" literals

### ISSUE THREAD

**Issue by jippi** [hashicorp/hcl2#20]
https://blog.golang.org/strings "UTF-8 and string literals"

```
derp = `${hello}`  # should be a string where "${hello}" is the value
```

Basically a way to disable all interpolation and processing, everything between the "`"  should be as-in, unprocessed

**Comment by apparentlymart** [comment]
Hi @jippi,

This is a _little_ different than Go, because Go doesn't have interpolation sequences to begin with, but I take your point here. At the moment we specifically detect attempts to use backticks and produce an explicit error, so that indicates that there's no ambiguity reason why we couldn't add this, and it just comes down to how complex it makes the parser. I'll take a look when I get a chance.

You say "a ton of HCL sensitive characters"... I'm curious to see an example of what you mean here, since HCL is only sensitive to a few special sequences: `${`, `%{` (new in HCL2) and backslash escapes. Does the hashi-helper program tend to encourage strings that contain these sequences more than your average (e.g.) Terraform configuration?

---

Interestingly, we intend for the JSON syntax to be easier to machine-generate than the native syntax but yet if we added this feature someone generating JSON would still need to deal with the escaping. I suppose in JSON use of this feature would look like the following:

```json
"${` anything you ${like} in here `}" => "anything you like in here"
```

...which isn't too bad, since the prefix and suffix are just hard-coded strings.

Another consideration is what we'd do about the muilti-line "heredoc" strings:

```hcl
foo = <<BAR
people commonly want to put ${shell-style-variables} in here too,
but we support template syntax in here and so it fails.
BAR
```

I suppose we could suggest that they just split a backtick string over multiple lines as people do in Go?

```hcl
foo = `
This could work but then you lose the convenient indentation handling of <-foo,
forcing everything to be flush with the left margin.
`
```

**Comment by jippi** [comment]
Yeah, hashi-helper got a lot of random passwords, so it happens that they include `$`, `${` and even `"` and `'` 

`"${` anything you ${like} in here `}" => "anything you ${like} in here"` would be my case, basically "don't do anything other than build a string until you see the next "`" (newlines or not) 

99% of my cases are single line cases though

**Comment by jippi** [comment]
Any plans to implement this?

**Comment by jippi** [comment]
Hi @apparentlymart 

Funny enough it's not for terraform usage, at @seatgeek we are using HCL + Go for a ton of things, for example [hashi-helper](https://github.com/seatgeek/hashi-helper) and a bunch of internal projects.

My ask is specifically for hashi-helper needs where we got vault secrets that sometimes contain things HCL try to interpret. Using HEREDOC usually fix it, but its visually noisy for a simple one-line string , which is why i wanted something like the go " ` " or PHP style single quote `'` meaning "just process this as a literal"

**Comment by tj** [comment]
I like the ticks in favour of heredocs quite a bit
